src/qibolab/_core/instruments/qm/config/devices.py

```diff
@@ -101,20 +107,32 @@ class OctaveInput:
 class Controller:
     type: ModuleTypes = "opx1"
     """https://docs.quantum-machines.co/latest/docs/Introduction/config/?h=opx10#controllers"""
-    analog_outputs: PortDict[dict[str, AnalogOutput]] = field(default_factory=PortDict)
-    digital_outputs: PortDict[dict[str, dict]] = field(default_factory=PortDict)
-    analog_inputs: PortDict[dict[str, AnalogInput]] = field(
-        default_factory=lambda: PortDict(DEFAULT_INPUTS)
+    analog_outputs: dict[int, dict] = field(default_factory=dict)
+    digital_outputs: dict[int, dict] = field(default_factory=dict)
+    analog_inputs: dict[int, Union[AnalogInput, MwFemInput]] = field(
+        default_factory=dict
     )
 
+    def _set_default_inputs(self):
+        """Add default inputs in controller config section.
+
+        Inputs are always registered to avoid issues with automatic mixer
+        calibration when using Octaves.
```

Comment:
Any reference to the potential issues?

Reply:
As far as I remember, it is more or less what this comment describes: inputs need to be always registered, even if not used, otherwise some mixer calibration script was failing. However, I also have not tried this for very long time (certainly 3+ months) so maybe it is no longer true.

The summary is: registering the inputs is potentially redundant but does not really hurt anything. Removing those will probably not affect any experiments, except the mixer calibration.